### PR TITLE
Proxy Implementation for Windows Chef Omnibus

### PR DESCRIPTION
**Pull request by afiune**
As a corporate end user I need Chef Omnibus for Windows to work behind a proxy so I can be able to use it on my company with proxy enabled.

### CODE REVIEW

File: support/chef_helpers.ps1 [hunk 1]
```diff
@@ -23,7 +23,11 @@ function should_update_chef($version) {
 # @param $chef_msi Temporal MSI path
 function download_chef($chef_url, $chef_msi) {
   Write-Host -NoNewline "`r`tDownloading Chef..."
-  (New-Object System.Net.WebClient).DownloadFile($chef_url, $chef_msi)
+  $proxy = new-object System.Net.WebProxy
+  $proxy.Address = $env:http_proxy
```
Comment (jdmundrawala) [inline support/chef_helpers.ps1]: Will this work if $env:http_proxy is null?

**Comment by jdmundrawala** [comment]
I think it might be nice to log that a proxy is being used

File: support/chef_helpers.ps1 (hunk 1 above)
Comment (afiune) [inline support/chef_helpers.ps1]: Yes it does. It is translated to: `there is no proxy` :smile:

**Comment by afiune** [comment]
\cc @fnichol @test-kitchen/windows

**Comment by tyler-ball** [comment]
:+1: 

@afiune @jdmundrawala If I understand the comment correctly, it will already output `there is no proxy` if  proxy is nil, correct?  That seems a good enough debug log.

**Comment by fnichol** [comment]
Didn't realize this needed to be explicit.

**Pull request merged**